src/paint.js

```diff
@@ -8,7 +8,10 @@ function setupPaint({ $, qlik }) {
 
   return function ($element, layout) {
     // Call irregularUtils to page the data for > 10000 points
-    const maxPages = qlik.navigation.getMode() === "analysis" ? 10 : 1;
+    const maxPages = this.inAnalysisState() ? 10 : 1;
```

Comment:
`this.inAnalysisState` is a function which is ran on-demand throughout the paint several times. Is there a risk of calling it once and read it from there for this file?

Reply:
you can store it at the top of the function, same as the options parameters